src/math.rs

```diff
@@ -216,3 +216,23 @@ pub fn cast4<T, U>(x: Point4<T>) -> Point4<U>
 {
     map4(x, cast)
 }
+
+pub mod interp {
+    use num_traits::Float;
+    use math;
+
+    /// Performs linear interploation between two values.
+    pub fn linear<T: Float>(a: T, b: T, x: T) -> T {
+        a + x * (b - a)
+    }
+
+    /// Maps a value onto a cubic S-curve.
+    pub fn s_curve3<T: Float>(x: T) -> T {
```

Comment:
What about `sigmoid3`? A link to https://en.wikipedia.org/wiki/Sigmoid_function might be cool too.